crates/polars-core/src/chunked_array/logical/categorical/mod.rs

```diff
@@ -89,6 +97,62 @@ impl CategoricalChunked {
         out
     }
 
+    pub fn to_global(&self) -> PolarsResult<Self> {
+        polars_ensure!(using_string_cache(), string_cache_mismatch);
+        // Fast path
+        let categories = match &**self.get_rev_map() {
+            RevMapping::Global(_, _, _) => return Ok(self.clone()),
+            RevMapping::Local(categories, _) => categories,
+            RevMapping::Enum(categories, _) => categories,
+        };
+        let physical = self.physical();
+
+        let mut builder = CategoricalChunkedBuilder::new(self.name(), physical.len());
+        builder.global_map_from_local(physical, categories.clone());
+        Ok(builder.finish())
+    }
+
+    pub fn to_enum(&self, categories: &Utf8Array<i64>, hash: u128) -> PolarsResult<Self> {
```

Comment:
Maybe add a docstring here. Explaining that we only allow extending `[a, b] -> [a, b, c]` or reordering `[a, b] -> [b, a]` enums here.